fairseq/training.py

```diff
@@ -231,9 +244,28 @@ def create_fairseq_hydra_config(cls, fairseq_hydra_config, max_epoch, max_update
         }
         res.update(FairseqHydraConfig(config_dict, post_config_dict))
         return res
+    
+    def _fairseq_prepare_checkpoint(self, start_checkpoint):
+        # rename the start checkpoint to checkpoint_last.pt if it is not None and checkpoint_last.pt does not exist
+        if start_checkpoint is None:
+            print("No start checkpoint provided")
+            return
+        if not os.path.exists(start_checkpoint):
+            raise FileNotFoundError(f"Start checkpoint {start_checkpoint} does not exist")
+        if not os.path.exists(os.path.join(self.out_checkpoint_dir.get_path(), "checkpoint_last.pt")):
```

Comment:
I don't think `checkpoint_last.pt` can ever exist here, because we call this function only once in the `create_files` Task.
When we resume a training (i.e. Task `run`) later, this function is not called again.
((but we can still check 🤷 ))

Reply:
Thanks for that hint, I didn't think about that `create_files` will be only called once. However, I would still argue for leaving that check in there to avoid the file being rewritten and therefore losing the progress of previous trainings in the form of the `checkpoint_last.pt` file. IMO it doesn't hurt code comprehension/readability and only makes the intention of the code more explicit